transcoder/docs/playback.md

```diff
@@ -0,0 +1,178 @@
+
+## Playback
+
+### Playlist requests
+```mermaid
+sequenceDiagram
+    participant cvp as Client video player
+    box Transcoder service (1..N)
+        participant api as Web API
+        participant jobs as Job worker
+    end
+    box Backend (HA)
+        participant db as Postgres
+        participant fs as Storage
+    end
+
+    cvp ->> api: Request playlist
+    api ->> db:  Get video metadata (segmentation times)
+    db -->> api: Return result
+    critical Metadata not available
+        api -->> api: Generate metadata
+        api -) db: Cache metadata
+    end
+    api ->> api: Generate playlist
+    api -) db: Create transcoding job for first k segments
+    api ->> cvp: Return video playlist
+```
+
+### Segment requests
+```mermaid
+sequenceDiagram
+    participant cvp as Client video player
+    box Transcoder service (1..N)
+        participant api as Web API
+        participant jobs as Job worker
+    end
+    box Backend (HA)
+        participant db as Postgres
+        participant fs as Storage
+    end
+
+    cvp ->> api: Requests video segment
+    loop Until segment is available
+        critical Get segment URL
+            api ->> db: Request segment URL (worker, S3)
+        option Segment exists, not pending deletion
+            db ->> db: Update segment access time<br/>via trigger (for cleanup)
+            db -->> api: Return URL
+        option Segment pending deletion
+            db -->> api: Return not available
+        option Segment does not exist, job not in progress
+            db ->> db: Create transcoding job for k segments
+            db -->> api: Return not available
+        end
+    end
+    api ->> storage: Request segment
+    storage -->> api: Return segment
+    api -->> cvp: Return segment
+```
+
+### Segment cleanup
+```mermaid
+sequenceDiagram
+    participant cvp as Client video player
+    box Transcoder service (1..N)
+        participant api as Web API
+        participant jobs as Job worker
+    end
+    box Backend (HA)
+        participant db as Postgres
+        participant fs as Storage
+    end
+
+    loop pg_cron: trigger every time duration d
+        db ->> db: Create segment cleanup job
+        worker ->> storage: Get all segments
+        storage -->> worker: Return segments
+        loop For each segment
+            critical Cleanup old segments
+                worker ->> db: Get last accessed time
+            option No record, segment older than expiration time t, or<br/>Record exists, segment access time older than expiration time t
+                worker ->> db: Mark segment as "pending deletion"
+                worker ->> storage: Delete segment
+                worker ->> db: Delete segment record
+            end
+        end
```

Comment:
This looks super complex for what it is. I don't think we need a pg_cron (which is an extension that is a pain to install) nor do we need a specific worker for that.

Reply:
Yea you're right. I was on the fence about whether or not this should be triggered by the DB, or if workers should have a long running ticker/goroutine that handles cleanup.

Reply:
I would keep the current goroutine/timer workflow, let's keep it simple